bin/ruby-build

```diff
@@ -394,6 +396,16 @@ fetch_tarball() {
   } >&4 2>&1
 }
 
+find_extracted_directory() {
+  for f in *; do
+    if [ -d "$f" ]; then
+      echo "$f"
+      return
+    fi
+  done
+  echo "Extracted directory not found"
```

Comment:
This failure message should go to `>&2`, and the function should `return 1` to make it exit with a non-zero status

Reply:
Won't the return value be ignored since this is called like
`extracted_dir="$(find_extracted_directory)"` ?
Is there a good pattern for dealing with this (checking return value and returning a string)?

Reply:
Actually returning 1 works as expected and fails the build :smiley: 
```
...
Extracted directory not found

BUILD FAILED (Fedora 26 using ruby-build 20180618-15-g4ff9ea3)
```

Reply:
Interestingly this doesn't error out if the caller is
```bash
local extracted_dir=$(find_extracted_directory)
```
but it works if declaration and assignment of `extracted_dir` are separate.
Bash is fun.